test/ResolutionManager.ts

```diff
@@ -1610,4 +1570,97 @@ describe("Resolution", async () => {
       );
     });
   });
+
+  describe("distrust resolution", async () => {
+    let totalVotingPower = 100;
+    async function _prepare() {
+      await resolution
+        .connect(managingBoard)
+        .createDistrustResolution("test", 6, false, [], [], user2.address);
+
+      voting.getTotalVotingPowerAt
+        .whenCalledWith(resolutionSnapshotId)
+        .returns(totalVotingPower);
+
+      await resolution.connect(managingBoard).approveResolution(resolutionId);
+      const approveTimestamp = await getEVMTimestamp();
+      await setEVMTimestamp(approveTimestamp + 3 * DAY);
+    }
+
+    it("should not remove and re-add delegation when not delegating", async () => {
+      await resolution
+        .connect(managingBoard)
+        .createDistrustResolution("test", 0, false, [], [], user2.address);
+
+      voting.getDelegate.whenCalledWith(user2.address).returns(user2.address);
+
+      await resolution.connect(managingBoard).approveResolution(resolutionId);
+
+      expect(voting.delegateOnBehalf).to.have.not.been.called;
```

Comment:
I tend to use a simpler form, in that case it would be `expect(voting.delegateOnBehalf).not.called`. Anyway that's not very important.